pkg/hwinfo/hwinfo_darwin.go

```diff
@@ -64,3 +66,52 @@ func listDisplays() ([]*monitorInfo, error) {
 	}
 	return result, nil
 }
+
+func listCPUs() ([]cpuInfo, error) {
+	var ret []cpuInfo
+	sysctl, err := exec.LookPath("/usr/sbin/sysctl")
+	if err != nil {
+		return ret, err
+	}
+
+	out, err := RunCommandWithContext(context.Background(), sysctl, "machdep.cpu")
+	if err != nil {
+		return nil, err
+	}
+
+	c := cpuInfo{}
+	for _, line := range strings.Split(string(out), "\n") {
+		values := strings.Fields(line)
+		if len(values) < 1 {
+			continue
+		}
+
+		if strings.HasPrefix(line, "machdep.cpu.vendor") {
```

Comment:
Can you convert this to a `switch` statement?

Reply:
done